File: rest-service/src/main/java/mb/api/model/IOMultipartJobRequest.java
package mb.api.model;

import java.io.File;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import mb.api.model.io.JsonKeys;

@JsonDeserialize(as = IOMultipartJobRequestImpl.class)
public interface IOMultipartJobRequest
{
  @JsonProperty(JsonKeys.Query)
  File getQuery();

  @JsonProperty(JsonKeys.Query)
  void setQuery(File query);

  @JsonProperty(JsonKeys.Properties)
  IOJsonJobRequest getProperties();

  @JsonProperty(JsonKeys.Properties)
  void setProperties(IOJsonJobRequest properties);
}

File: rest-service/src/main/java/mb/api/model/IOMultipartJobRequest.kt
package mb.api.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonProperty
import mb.api.model.io.JsonKeys
import java.io.File

@JsonInclude(NON_NULL)
data class IOMultipartJobRequest(
  @JsonProperty(JsonKeys.Query)
  var query: File? = null,

  @JsonProperty(JsonKeys.Properties)
  var properties: IOJsonJobRequest? = null,
)